packages/client/modules/invoice/components/Invoice/Invoice.tsx

```diff
@@ -174,10 +173,11 @@ const Invoice = (props: Props) => {
     tier
   } = invoiceDetails
   const status = invoiceDetails.status as InvoiceStatusEnum
-  const {interval, nextPeriodEnd} = nextPeriodCharges!
+  const {amount, interval, nextPeriodEnd} = nextPeriodCharges!
   const chargeDates = `${makeDateString(startAt)} to ${makeDateString(endAt)}`
   const nextChargesDates = `${makeDateString(endAt)} to ${makeDateString(nextPeriodEnd)}`
-  const discountedAmount = coupon ? `-${coupon.amountOff && invoiceLineFormat(coupon.amountOff) || coupon.percentOff && invoicePercentToDiscountedAmount(nextPeriodCharges.amount, coupon.percentOff)}` : ''
+  const amountOff = coupon && (coupon.amountOff || coupon.percentOff! / 100 * amount)
```

Comment:
@mattkrick beautiful—what’s the bang `!` after percentOff do?

Reply:
I guess this: https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

Reply:
TIL: non-null assertion operator

Reply:
yep! typescript doesn't offer mutually exclusive conditions, but _we_ know that if percentOff doesn't exist, amountOff does & vice versa, so it's safe to assume it's non null